internal/speakerlist/speakerlist.go

```diff
@@ -27,6 +27,7 @@ import (
 
 // SpeakerList represents a list of healthy speakers.
 type SpeakerList struct {
+	sync.Mutex
```

Comment:
Embedding like this means the godoc for this type will now have public Lock and Unlock methods. Are you okay with that?

(I don't care a huge amount, but that's the usual rationale for having explicit mutex fields)

Also, please document what this mutex covers. Empirically it seems to only cover mkSpeakerIPs, but its position in the type suggests it protects all fields.

Reply:
Added a comment, `// Must be locked while accessing any of the internal SpeakerList maps or arrays`